019_media_types is flaky again

Experienced here https://github.com/denoland/deno/pull/4035/checks?check_run_id=457778529 but I've also seen elsewhere.

I think it might be related to the changes to the test server in 2e7d449623b7faee8feb0a2970e0d6bb7c66b409. Or maybe the changes in e6167c78134182c45689bda7bcb12af05009349c.

The same cause but different failure https://github.com/denoland/deno/runs/465322195

```
2020-02-24T18:11:14.8894330Z failures:
2020-02-24T18:11:14.8894610Z 
2020-02-24T18:11:14.8895870Z ---- _049_info_flag_script_jsx stdout ----
2020-02-24T18:11:14.8896350Z target_dir /Users/runner/runners/2.165.2/work/deno/deno/target/release
2020-02-24T18:11:14.8896760Z root path /Users/runner/runners/2.165.2/work/deno/deno/cli/..
2020-02-24T18:11:14.8897160Z deno_exe path /Users/runner/runners/2.165.2/work/deno/deno/target/release/deno
2020-02-24T18:11:14.8897590Z target_dir /Users/runner/runners/2.165.2/work/deno/deno/target/release
2020-02-24T18:11:14.8897960Z OUTPUT
2020-02-24T18:11:14.8898270Z 
2020-02-24T18:11:14.8898600Z OUTPUT
2020-02-24T18:11:14.8919140Z ##[error]thread '_049_info_flag_script_jsx' panicked at 'bad exit code, expected: 0, actual: 1', cli/tests/integration_tests.rs:1264:9
2020-02-24T18:11:14.8926160Z error: test failed, to rerun pass '-p deno --test integration_tests'
2020-02-24T18:11:14.8926390Z stack backtrace:
2020-02-24T18:11:14.8927430Z    0:        0x104e06005 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h109831b4e69a3cdc
2020-02-24T18:11:14.8928610Z    1:        0x104e29900 - core::fmt::write::haefb4093ced4abcf
2020-02-24T18:11:14.8929860Z    2:        0x104b935b9 - std::io::Write::write_fmt::h57d9d260264ecc5b
2020-02-24T18:11:14.8931190Z    3:        0x104e002ac - std::io::impls::<impl std::io::Write for alloc::boxed::Box<W>>::write_fmt::hb0f0233a36cd2ce6
2020-02-24T18:11:14.8932860Z    4:        0x104e08473 - std::panicking::default_hook::{{closure}}::hd11104060d683e7f
2020-02-24T18:11:14.8934200Z    5:        0x104e08127 - std::panicking::default_hook::h3ac89e0bbec67847
2020-02-24T18:11:14.8935460Z    6:        0x104e08b1b - std::panicking::rust_panic_with_hook::h497844b5b7d1708e
2020-02-24T18:11:14.8936180Z    7:        0x104e086b9 - rust_begin_unwind
2020-02-24T18:11:14.8937430Z    8:        0x104e318ff - std::panicking::begin_panic_fmt::hc560df0756655667
2020-02-24T18:11:14.8941210Z    9:        0x104b8ddd8 - integration_tests::util::CheckOutputIntegrationTest::run::h958a8016b3a63d4b
2020-02-24T18:11:14.8942520Z   10:        0x104b809bd - core::ops::function::FnOnce::call_once::h0c9a232dee04cf63
2020-02-24T18:11:14.8943870Z   11:        0x104b9e6fe - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h44c450ea090845aa
2020-02-24T18:11:14.8945490Z   12:        0x104e10d4f - __rust_maybe_catch_panic
2020-02-24T18:11:14.8946470Z   13:        0x104bb7fc5 - test::run_test::run_test_inner::{{closure}}::h433eeed657bd4954
2020-02-24T18:11:14.8947360Z   14:        0x104b92a7b - std::sys_common::backtrace::__rust_begin_short_backtrace::h06b79d7b3e88f6bd
2020-02-24T18:11:14.8948190Z   15:        0x104b96f4b - std::panicking::try::do_call::hcb5b9f07da0d5a0c
2020-02-24T18:11:14.8948700Z   16:        0x104e10d4f - __rust_maybe_catch_panic
2020-02-24T18:11:14.8949570Z   17:        0x104b97926 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h01ec0e0a5cf97627
2020-02-24T18:11:14.8950450Z   18:        0x104dfbe7e - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::hf89a561d6d706029
2020-02-24T18:11:14.8951310Z   19:        0x104e1035e - std::sys::unix::thread::Thread::new::thread_start::hd09405f1ea9f0a0f
2020-02-24T18:11:14.8951820Z   20:     0x7fff66080e65 - _pthread_start
2020-02-24T18:11:14.8951900Z 
2020-02-24T18:11:14.8951980Z 
2020-02-24T18:11:14.8952100Z failures:
2020-02-24T18:11:14.8952230Z     _049_info_flag_script_jsx
2020-02-24T18:11:14.8952310Z 
2020-02-24T18:11:14.8952460Z test result: FAILED. 121 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
2020-02-24T18:11:14.8952550Z 
```

--- a/cli/test_util.rs
+++ b/cli/test_util.rs
@@ -67,39 +67,32 @@ fn kill_http_server() {
 
 pub fn http_server() -> HttpServerGuard {
   SERVER_COUNT.fetch_add(1, Ordering::SeqCst);
-
   {
     let mut server_guard = SERVER.lock().unwrap();
     if server_guard.is_none() {
       println!("tools/http_server.py starting...");
       let mut child = Command::new("python")
         .current_dir(root_path())
-        .args(&[
-          "-u",
-          "tools/http_server.py",
-          "--certfile",
-          root_path()
-            .join("std/http/testdata/tls/localhost.crt")
-            .to_str()
-            .unwrap(),
-          "--keyfile",
-          root_path()
-            .join("std/http/testdata/tls/localhost.key")
-            .to_str()
-            .unwrap(),
-        ])
+        .args(&["-u", "tools/http_server.py"])
         .stdout(Stdio::piped())
         .spawn()
         .expect("failed to execute child");
 
       let stdout = child.stdout.as_mut().unwrap();
       use std::io::{BufRead, BufReader};
-      let mut lines = BufReader::new(stdout).lines();
-      let line = lines.next().unwrap().unwrap();
-      assert!(line.starts_with("ready"));
-      server_guard.replace(child);
+      let lines = BufReader::new(stdout).lines();
+      // Wait for "ready" on stdout. See tools/http_server.py
+      for maybe_line in lines {
+        if let Ok(line) = maybe_line {
+          if line.starts_with("ready") {
+            server_guard.replace(child);
+            break;
+          }
+        } else {
+          panic!(maybe_line.unwrap_err());
+        }
+      }
     }
   }
-
   HttpServerGuard {}
 }

--- a/tools/http_server.py
+++ b/tools/http_server.py
@@ -11,7 +11,6 @@ import socket
 import sys
 from time import sleep
 from threading import Thread
-from threading import Lock
 from util import root_path
 import ssl
 import getopt

--- a/tools/http_server.py
+++ b/tools/http_server.py
@@ -11,6 +11,7 @@ import socket
 import sys
 from time import sleep
 from threading import Thread
+from threading import Lock
 from util import root_path
 import ssl
 import getopt
@@ -21,14 +22,11 @@ REDIRECT_PORT = 4546
 ANOTHER_REDIRECT_PORT = 4547
 DOUBLE_REDIRECTS_PORT = 4548
 INF_REDIRECTS_PORT = 4549
-
 HTTPS_PORT = 5545
 
 
 def create_http_arg_parser():
     parser = argparse.ArgumentParser()
-    parser.add_argument('--certfile')
-    parser.add_argument('--keyfile')
     parser.add_argument('--verbose', '-v', action='store_true')
     return parser
 
@@ -36,8 +34,8 @@ def create_http_arg_parser():
 HttpArgParser = create_http_arg_parser()
 
 args, unknown = HttpArgParser.parse_known_args(sys.argv[1:])
-CERT_FILE = args.certfile
-KEY_FILE = args.keyfile
+CERT_FILE = os.path.join(root_path, "std/http/testdata/tls/localhost.crt")
+KEY_FILE = os.path.join(root_path, "std/http/testdata/tls/localhost.key")
 QUIET = not args.verbose
 
 
@@ -314,10 +312,19 @@ def start(s):
 def spawn():
     servers = (server(), redirect_server(), another_redirect_server(),
                double_redirects_server(), https_server())
-    while any(not s.thread.is_alive() for s in servers):
-        sleep(0.01)
+    # In order to wait for each of the servers to be ready, we try connecting to
+    # them with a tcp socket.
+    for running_server in servers:
+        client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        port = running_server.server.server_address[1]
+        client.connect(("127.0.0.1", port))
+        print "connected", port
+        client.close()
+        assert running_server.thread.is_alive()
+    # The following output "ready" is specificly looked for in cli/test_util.rs
+    # to prevent race conditions.
+    print "ready"
     try:
-        print "ready"
         yield servers
     finally:
         for s in servers:

